### PR TITLE
Integration tests using spectron

### LINKED ISSUE #398: Integration tests
We should add integration test framework that is used for GUI testing (and run within CI). Spectron (https://github.com/electron-userland/spectron) seems to be one of the most popular frameworks for Electron.

### PR DESCRIPTION
**Pull request by jakolehm**
Fixes #398

### CODE REVIEW

File: .azure-pipelines.yml [hunk 1]
```diff
@@ -119,6 +121,10 @@ jobs:
         displayName: Lint
       - script: make test
         displayName: Run tests
+      # - script: sudo apt-get update && sudo apt-get install libgconf-2-4 -y
+      #   displayName: Install integration test dependencies
+      # - script: xvfb-run --auto-servernum --server-args='-screen 0, 1600x900x24' make integration-linux
```
Comment (jakolehm) [inline .azure-pipelines.yml]: For some reason I cannot make this work on Linux.. tried also on my own VM.

**Comment by jakolehm** [comment]
@nevalla @jnummelin PTAL

File: spec/integration/specs/app_spec.ts [hunk 2]
```diff
@@ -0,0 +1,56 @@
...
+  }, 20000)
+
+  it('shows "whats new"', async () => {
+    await clickWhatsNew(app)
+  })
+
+  it('allows to add a cluster', async () => {
+    const status = spawnSync("minikube status", {shell: true})
+    if (status.status !== 0) {
+      console.warn("skipping test")
```
Comment (nevalla) [inline spec/integration/specs/app_spec.ts]: Should we add some reason here?

**Review by nevalla — APPROVED** [review]
Added minor comment, otherwise LGTM

File: spec/integration/specs/app_spec.ts (hunk 2 above)
Comment (jakolehm) [reply spec/integration/specs/app_spec.ts]: Yes, added.

**Review by jnummelin — APPROVED** [review]
LGTM

Just wondering if there's some specific reason not to enable integration tests on Windows pipeline? Can be done in separate PR, need to remember to add an issue on that.

**Comment by jakolehm** [comment]
> Just wondering if there's some specific reason not to enable integration tests on Windows pipeline? Can be done in separate PR, need to remember to add an issue on that.

Didn't have time to test this on Windows. I'll create a separate issue about it.

**Pull request merged**